utils/stm-to-jitx.stanza

```diff
@@ -42,30 +43,36 @@ public defn to-jitx-pin-properties (pin-properties:STMPinProperties) -> False :
     if not numeric-pads? and not named-pads? :
       fatal("Generated a micro-controller without pads.") 
     
+    ; We need to compile a list of reused pad to pin mappings.
+    val pad-mappings = HashTable<String|Int, Vector<String>>(Vector<String>)
+    defn check-pad (pad:String|Int, pin:String) : 
+      add(pad-mappings[pad], pin)
+      length(pad-mappings[pad]) == 1
+
     ;Generate the pin-properties.
     ;Cases are split between Int and Ref pads, and across generic and power properties usages.
-    if numeric-pads? :
-      pin-properties :
-        [pin:Ref | pads:Int ... | side:Dir | generic-pin:GenericPin | power-pin:PowerPin]
-        for row in rows(pin-properties) do :
-          val pin-ref = ref-string-to-ref(pin(row))
-          val pad-index = to-int(pad(row) as Double)
-          match(generic-props?(row), power-props?(row)) :
-            (g:True, p:True)   : [(pin-ref) | pad-index | side(row) | generic-props | power-props]
-            (g:True, p:False)  : [(pin-ref) | pad-index | side(row) | generic-props |      -     ]
-            (g:False, p:True)  : [(pin-ref) | pad-index | side(row) |       -       | power-props]
-            (g:False, p:False) : [(pin-ref) | pad-index | side(row) |       -       |      -     ]
-    else :
-      pin-properties :
-        [pin:Ref | pads:Ref ... | side:Dir | generic-pin:GenericPin | power-pin:PowerPin]
-        for row in rows(pin-properties) do :
-          val pin-ref = ref-string-to-ref(pin(row))
-          val pad-name = raw-string-to-ref(pad(row) as String)
+    pin-properties : 
+      [pin:Ref | pads : (Ref|Int) ... | side:Dir | generic-pin:GenericPin | power-pin:PowerPin]
+      for row in rows(pin-properties) do : 
+        val pin-ref = ref-string-to-ref(pin(row))
+        val pad-ref = match(pad(row)):
+          (i:Int): i
+          (s:String): raw-string-to-ref(s)
+
+        if check-pad(pad(row), pin(row)) : 
```

Comment:
So we silently ignore when several pins are mapped to the same pad? Why does this happen in the first place?

Reply:
We're not ignoring it, it's possible for multiple pin names to refer to the same schematic pin. We pick the first we come across as the canonical pin. 

Ideally we would rename, but that is a bit trickier given the naming conventions so we can fix that in a subsequent PR